### ISSUE TITLE
Can accidentally flood channel when pasting one very long line.

### ISSUE THREAD

**Issue by barnaba** [weechat/weechat#1117]
When pasting one long line without any newlines the warning about pasting multiple lines doesn't show up, even though weechat will break the line into multiple lines.

How to reproduce:
1. copy
> Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed ex dui, consectetur lacinia arcu in, ultrices dignissim nulla. Nam sed vehicula eros, vel maximus ipsum. Donec et blandit enim, ac vehicula sem. Integer pharetra, urna a pretium scelerisque, est odio tincidunt felis, vestibulum tincidunt sapien sapien cursus magna. Suspendisse ut massa molestie, tincidunt magna ut, ultrices elit. Morbi ut felis sit amet lectus placerat molestie ut vitae ipsum. Praesent euismod purus eget quam aliquet porttitor. Praesent venenatis massa quis tempus accumsan. Morbi condimentum tincidunt nibh, ut pulvinar nibh vestibulum id. Vivamus a elit imperdiet justo maximus convallis vitae et metus. Mauris at scelerisque purus. Vestibulum sagittis condimentum blandit. Fusce et nibh ut urna porta sagittis at eu dui. Nulla auctor ac nibh id egestas. Cras leo ante, laoreet a ante ut, maximus sagittis risus. Aliquam fringilla et nulla et pellentesque. Aenean porta ultrices quam, vitae interdum odio suscipit eget. Suspendisse euismod nunc lacus, vitae sagittis ante finibus sed. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Mauris eget purus quis turpis laoreet feugiat. Vivamus ac nisi vitae enim vestibulum semper ac in lacus. Donec molestie dui magna, nec consectetur eros laoreet a. Aliquam nec arcu lacinia est ullamcorper efficitur eu vel leo. Proin eget eros bibendum, pellentesque erat at, gravida libero. Cras id consectetur leo, eget laoreet diam. Proin sit amet dictum quam. Pellentesque facilisis vitae diam non auctor. Proin porttitor urna sit amet velit semper, eget suscipit magna imperdiet. Fusce feugiat ipsum et risus tempus maximus. Curabitur sit amet tellus lorem. Fusce et scelerisque erat. Aenean tincidunt ut massa finibus tristique. Pellentesque quis metus dignissim diam malesuada lacinia.Proin auctor eros libero, rhoncus dapibus felis egestas eu. Vivamus ultrices, elit vel dapibus fermentum, nisi quam accumsan ligula, quis placerat arcu quam id metus. Nunc egestas mi eu lorem finibus, vitae cursus lorem pharetra. Etiam ornare volutpat nisl. Nulla nisi velit, porttitor eu aliquet vitae, dictum at tellus. Donec pretium condimentum lectus tincidunt auctor. Curabitur ornare tortor ligula, nec elementum magna luctus quis. Proin ultricies mauris in augue varius imperdiet. Ut non luctus tortor. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Vestibulum quis scelerisque magna.
2. paste into a channel or primsg buffer

Current behaviour: You flood the channel

Expected behaviour: client should calculate the actual number of lines that it is going to send and display a prompt: "[Paste 7 lines ? [ctrl-Y] Yes [ctrl-N] No]"

**Comment by flashcode** [comment]
Hi,
The problem is that the paste warning is made by core (not by irc plugin), and the message is split by irc plugin after this message is displayed.
By design, the core doesn't know if some plugin will split the message before it is sent.
To "fix" that, some things must be changed, but that's not easy: the irc plugin splits into multiple message but this depends on the context (server, length of nick, ...).
So maybe a new option to set a line limit, so that the paste will consider the line is in fact 2 or N lines, according to the option?

**Comment by barnaba** [comment]
> So maybe a new option to set a line limit, so that the paste will consider the line is in fact 2 or N lines, according to the option?

My personal view is not many people would use that option and it'll just add complexity to the code (and documentation) for little benefit. There's already a lot of options making discovery of new ones hard. I certainly would use the option if it was implemented, but I doubt many people will ever learn about it.

I understand that it's not problematic enough to warrant a redesign of such a scale. It's probably fine if it just stays as a consideration for any future refactoring that might be required to solve any more serious problems.

Thanks for a quick reply and explaining the situation.

**Comment by flashcode** [comment]
Yes an option for a warning when pasting long lines can be added, good idea.
The warning could then be: `[Paste 2 lines (9000 characters)? [ctrl-Y] Yes [ctrl-N] No]`.

**Comment by bradyt** [comment]
Accidentally pasted a long image url, hoping `/set weechat.bar.input.size 0` will help avoid that in the future.
